app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt

```diff
@@ -0,0 +1,192 @@
+  while (current <= end) {
+    dates.add(current)
+    current = current.plusDays(1)
+  }
+  return dates
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun WeatherCarousel(weatherData: List<Weather>) {
+  val pagerState = rememberPagerState { weatherData.size }
+
+  Column {
+    HorizontalPager(
+        state = pagerState,
+        modifier = Modifier.fillMaxWidth(),
+        beyondBoundsPageCount = weatherData.size - 1) { page ->
+          WeatherCard(weatherData[page])
+        }
+
+    Row(modifier = Modifier.align(Alignment.CenterHorizontally).padding(16.dp)) {
+      repeat(weatherData.size) { iteration ->
+        val color =
+            if (pagerState.currentPage == iteration) MaterialTheme.colorScheme.primary
+            else Color.LightGray
+        Box(modifier = Modifier.padding(2.dp).clip(CircleShape).background(color).size(10.dp))
+      }
+    }
+  }
+}
+
+@Composable
+fun WeatherCard(weather: Weather) {
+  val configuration = LocalConfiguration.current
+  val locale = configuration.locales[0]
+  val formatter = DateTimeFormatter.ofPattern("EEEE dd MMMM yyyy", locale)
+  val formattedDate = weather.date.format(formatter)
+
+  Card(
+      modifier = Modifier.fillMaxWidth().padding(horizontal = 10.dp).testTag("weather card"),
+      shape = MaterialTheme.shapes.medium) {
+        Column(
+            modifier = Modifier.padding(10.dp).fillMaxWidth(),
+            horizontalAlignment = Alignment.Start,
+            verticalArrangement = Arrangement.spacedBy(8.dp)) {
+              Text(
+                  text = formattedDate,
+                  style = ChimpagneTypography.titleSmall.copy(fontWeight = FontWeight.Bold))
+              Row(
+                  verticalAlignment = Alignment.CenterVertically,
+                  horizontalArrangement = Arrangement.spacedBy(16.dp)) {
+                    Box(modifier = Modifier.size(56.dp).clip(CircleShape)) {
+                      Image(
+                          painter = rememberAsyncImagePainter("https:" + weather.weatherIcon),
+                          contentDescription = null,
+                          modifier = Modifier.size(56.dp))
+                    }
+                    Column {
+                      Text(
+                          modifier = Modifier.testTag("weather temp"),
+                          text =
+                              "Temp : ${weather.temperatureLow}°C -> ${weather.temperatureHigh}°C",
+                          style = ChimpagneTypography.bodyMedium)
+                      Text(
+                          modifier = Modifier.testTag("weather wind"),
+                          text =
+                              stringResource(R.string.wind_speed) +
+                                  " : ${weather.maxWindSpeed} km/h",
+                          style = ChimpagneTypography.bodyMedium)
+                    }
+                  }
+            }
+      }
+}
+
+@Composable
+fun TextMessage(message: String?) {
```

Comment:
Is it really useful to create a component inside another one ? It is used only one time